### PR TITLE
[stable/nats] Improve Prometheus Metrics

### PR DESCRIPTION
**Pull request by ttjiaa**
Signed-off-by: Tong Jia <ttjiaa@gmail.com>

#### This PR:
- improves Prometheus metrics sidecar:
  - updates `prometheus-nats-exporter` image to image hosted by `Synadia`
  - templates metrics `port`
  - templates `args` and replaces `command`
  - add additional `flags` under `args` to expose more metrics groups by default:
    - `-connz`: Get connection metrics
    - `-routez`: Get route metrics
    - `-subz`: Get subscription metrics
    - `-varz`: Get connection metrics
  - updates `NOTES.txt` to include instructions on accessing `/metrics`
  - updates `README.md`
- fixes typo in `README.md` for `statefulset.rollingUpdatePartition`
      

#### Checklist

- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/nats/Chart.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: nats
-version: 1.2.0
+version: 1.2.1
```
Comment (juan131) [inline stable/nats/Chart.yaml]: This PR breaks backwards compatibility. Therefore, it should increase the major version of the chart.

File: stable/nats/values.yaml [hunk 2]
```diff
@@ -250,36 +250,45 @@ ingress:
...
+  ## Metrics exporter resource requests and limits
   ## ref: http://kubernetes.io/docs/user-guide/compute-resources/
   ##
   # resources: {}
-   ## Metrics exporter pod Annotation
-  podAnnotations:
+  ## Metrics exporter port
+  port: 7777
+  ## Metrics exporter annotations
+  annotations:
```
Comment (juan131) [inline stable/nats/values.yaml]: Why did you change it from `podAnnotations` to `annotations`? Please note that this kind of changes will break backwards compatibility and you won't be able to upgrade the chart from previous version by running:

```
helm upgrade ...
```

File: stable/nats/values.yaml [hunk 3]
```diff
@@ -250,36 +250,45 @@ ingress:
...
+## Metrics / Prometheus NATS Exporter
 ##
+## ref: https://github.com/nats-io/prometheus-nats-exporter
 metrics:
   enabled: false
   image:
     registry: docker.io
-    repository: appcelerator/prometheus-nats-exporter
-    tag: 0.17.0
+    repository: synadia/prometheus-nats-exporter
```
Comment (juan131) [inline stable/nats/values.yaml:260]: What are the main advantages of using this new exporter?

**Review by juan131 — COMMENTED** [review]
Thanks for this contribution! Please take a look to my comments

File: stable/nats/values.yaml (hunk 3 above)
Comment (ttjiaa) [reply stable/nats/values.yaml:260]: I see the main advantages as:
- using the "official" dockerhub repo maintained by Synadia
- having an image tag that corresponds to the [`0.1.0`](https://github.com/nats-io/prometheus-nats-exporter/releases/tag/v0.1.0) release of `prometheus-nats-exporter`

File: stable/nats/Chart.yaml (hunk 1 above)
Comment (ttjiaa) [reply stable/nats/Chart.yaml]: I reverted the `podAnnotations` -> `annotations` change, the PR should now no longer contain breaking changes. I'll bump the minor, as the changes are more than bug fixes. Please let me know if this is ok.

Comment (juan131) [reply stable/nats/Chart.yaml]: cool! Thanks

File: stable/nats/values.yaml (hunk 3 above)
Comment (juan131) [reply stable/nats/values.yaml:260]: It seems it's the official one: https://github.com/nats-io/nats-operator/tree/master/helm/nats-operator

I'm okay with replacing it!
